src/audio/copier/copier.c

```diff
@@ -363,16 +363,6 @@ static int init_dai(struct comp_dev *parent_dev,
 	if (ret < 0)
 		goto e_zephyr_free;
 
-	cd->converter[IPC4_COPIER_GATEWAY_PIN] =
-			get_converter_func(&copier->base.audio_fmt, &copier->out_fmt, type,
-					   IPC4_DIRECTION(dai->direction));
```

Comment:
formats cannot change between multiple calls to `init_dai()`?

Reply:
correct, so the last patch moves it out to create_dai() instead